### PR TITLE
raftstore: support dr auto sync

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?

Support initializing DrAutoSync replication mode. In such case, logs
will not be committed before they are replicated to at least two
nodes with different label values. It utilizes raft's group commit
algorithm under the hook.

Dynamically change will be supported in next PR.

Part of #7097

### Check List 

Tests 

- Unit test
- Integration test

### Release note

* No release note

### CODE REVIEW

File: tests/integrations/raftstore/test_replication_mode.rs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    must_get_none(&cluster.get_engine(2), b"k1");
+    cluster.clear_send_filters();
+    must_get_equal(&cluster.get_engine(2), b"k1", b"v0");
+
+    pd_client.must_remove_peer(1, new_peer(3, 3));
+    must_get_none(&cluster.get_engine(3), b"k1");
+    cluster.add_send_filter(IsolationFilterFactory::new(3));
+    pd_client.must_add_peer(1, new_learner_peer(3, 5));
+    let region = cluster.get_region(b"k1");
+    // Peer 2 can be promoted as there will be enough quorum alive.
```
Comment (gengliqi) [inline tests/integrations/raftstore/test_replication_mode.rs:118]: Is it should be `Peer 5 can not be promoted...`?

File: tests/integrations/raftstore/test_replication_mode.rs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        res.get_header()
+            .get_error()
+            .get_message()
+            .contains("unsafe to perform conf change peer"),
+        "{:?}",
+        res
+    );
+}
+
+// Tests if group id is updated when adding new node and apply snapshot.
```
Comment (gengliqi) [inline tests/integrations/raftstore/test_replication_mode.rs:134]:
```suggestion
// Tests if group id is updated when adding new node and applying snapshot.
```

File: tests/integrations/raftstore/test_replication_mode.rs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        .insert("zone".to_owned(), "ES".to_owned());
+    cluster.run_node(1).unwrap();
+    cluster.run_node(2).unwrap();
+    cluster
+        .cfg
+        .server
+        .labels
+        .insert("zone".to_owned(), "WN".to_owned());
+    cluster.run_node(3).unwrap();
+    cluster.must_put(b"k1", b"v0");
```
Comment (gengliqi) [inline tests/integrations/raftstore/test_replication_mode.rs:158]: Why not use `prepare_cluster`?

Comment (BusyJay) [reply tests/integrations/raftstore/test_replication_mode.rs:158]: Need to change label value during setup.

**Pull request merged**